### PR TITLE
[TF FE] Avoid usage of NGraph routines

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Also, it introduces separate test fixture with graph comparison for FEs

**Ticket:** 111370

### CODE REVIEW

File: src/frontends/tensorflow/tests/convert_tricky_models.cpp [hunk 1]
```diff
@@ -235,11 +235,15 @@ TEST_F(TransformationTestsF, InjectedBodyAndIf) {
     }
 }
 
-TEST_F(TransformationTestsF, ModelWithDilatedGroupConvolution) {
+TEST_F(FrontEndConversionWithReferenceTestsF, ModelWithDilatedGroupConvolution) {
     {
         model = convert_model("dilated_gconv_model/dilated_gconv_model.pbtxt");
         // need to call MOC to fuse BatchToSpace/SpaceToBatch with GroupConvolution
         manager.register_pass<pass::MOCTransformations>(false);
+        // TODO: enable ATTRIBUTES, CONST_VALUES and ACCURACY checks, CVS-xxxx
```
Comment (itikhono) [inline src/frontends/tensorflow/tests/convert_tricky_models.cpp:243]: CVS-xxxx ??

Comment (rkazants) [reply src/frontends/tensorflow/tests/convert_tricky_models.cpp:243]: one moment)

**Review by gkrivor — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
https://dev.azure.com/openvinoci/dldt/_build/results?buildId=566772&view=logs&j=e0e676f4-9664-538a-6d6b-aa53516aa09f&t=57194276-3fc5-5e59-b843-57f10a631f8e&s=96ac2280-8cb4-5df5-99de-dd2da759617d

File: src/frontends/tensorflow/tests/convert_tricky_models.cpp [hunk 2]
```diff
@@ -375,7 +383,7 @@ TEST_F(TransformationTestsF, ModelWithMultioutputBodyGraphNode) {
     }
 }
 
-TEST_F(TransformationTestsF, ModelWithEmptyTensorListAndPushBack) {
+TEST_F(FrontEndConversionWithReferenceTestsF, ModelWithEmptyTensorListAndPushBack) {
```
Comment (ilya-lavrenov) [inline src/frontends/tensorflow/tests/convert_tricky_models.cpp:386]: https://dev.azure.com/openvinoci/dldt/_build/results?buildId=566772&view=logs&j=e0e676f4-9664-538a-6d6b-aa53516aa09f&t=57194276-3fc5-5e59-b843-57f10a631f8e&s=96ac2280-8cb4-5df5-99de-dd2da759617d

Comment (rkazants) [reply src/frontends/tensorflow/tests/convert_tricky_models.cpp:386]: Fixed by https://github.com/openvinotoolkit/openvino/pull/17727

**Pull request merged**